src/metadata.rs

```diff
@@ -75,10 +75,10 @@ pub enum MetadataError {
 /// Runtime metadata.
 #[derive(Clone, Debug)]
 pub struct Metadata {
-    modules: HashMap<String, ModuleMetadata>,
-    modules_with_calls: HashMap<String, ModuleWithCalls>,
-    modules_with_events: HashMap<String, ModuleWithEvents>,
-    modules_with_errors: HashMap<String, ModuleWithErrors>,
+    pub(crate) modules: HashMap<String, ModuleMetadata>,
```

Comment:
Prefer to add a constructor rather than making all these `pub(crate)`. Could probably just derive `Default` for your use case in the test.